maestro-test/src/test/kotlin/maestro/test/IntegrationTest.kt

```diff
@@ -2795,6 +2798,48 @@ class IntegrationTest {
         )
     }
 
+    @Test
+    fun `Case 103 - execute onFlowStart and onFlowComplete hooks`() {
+        val commands = readCommands("103_on_flow_start_complete_hooks")
+
+        val driver = driver {
+        }
+
+        val receivedLogs = mutableListOf<String>()
+
+        for (command in commands) {
+            val maestroCommand = command.asCommand()
+            if (maestroCommand is ApplyConfigurationCommand) {
+                Maestro(driver).use {
+                    val onFlowStart = maestroCommand.config.onFlowStart
+                    val onFlowComplete = maestroCommand.config.onFlowComplete
+                    if (onFlowStart != null) {
+                        orchestra(
+                            it,
+                            onCommandMetadataUpdate = { _, metadata ->
+                                receivedLogs += metadata.logMessages
+                            }
+                        ).runFlow(onFlowStart.commands)
+                    }
+                    if (onFlowComplete != null) {
+                        orchestra(
+                            it,
+                            onCommandMetadataUpdate = { _, metadata ->
+                                receivedLogs += metadata.logMessages
+                            }
+                        ).runFlow(onFlowComplete.commands)
+                    }
+                }
+            }
+        }
+
+        // Then
+        assertThat(receivedLogs).containsExactly(
+            "setup",
+            "teardown",
+        ).inOrder()
```

Comment:
Let's update this test to ensure that the `tapOn` command is being executed in between the setup / teardown scripts

Reply:
this order logic is done in MaestroCommandRunner, so it is impossible to test this scenario with IntegrationTest as for now.

we could add a tech debt ticket to make MaestroCommandRunner testable and have IntegrationTest scenarios wrapped around it.